src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java

```diff
@@ -83,11 +86,11 @@ private UacQidTuple fetchExistingUacQidPairsForAction(Case linkedCase, ActionTyp
       throw new RuntimeException(
           String.format("We can't process this case id '%s' with no UACs", caseId));
 
-    } else if (!actionType.equals(ActionType.ICHHQW)
+    } else if ((!actionType.equals(ActionType.ICHHQW) && !actionType.equals(ActionType.SPG_IC14))
```

Comment:
I would rewrite this whole chunk like this:

```
    if (uacQidLinks == null || uacQidLinks.isEmpty()) {
      throw new RuntimeException(
          String.format("We can't process this case id '%s' with no UACs", caseId));

    } else if (actionType == ActionType.ICHHQW || actionType == ActionType.SPG_IC14) {
      if (isStateCorrectForSecondWelshUacQidPair(linkedCase, uacQidLinks)) {
        return getUacQidTupleWithSecondWelshPair(uacQidLinks);
      }
    } else if (isStateCorrectForSingleUacQidPair(linkedCase, uacQidLinks)) {
      return getUacQidTupleWithSinglePair(uacQidLinks);
    }
    
    throw new RuntimeException(
        String.format("Wrong number of UACs for treatment code '%s'", actionType));
```

Reply:
Ok but why have you removed the second parameter from **_getUacQidTupleWithSecondWelshPair_**?

Reply:
that was a mistake

Reply:
Ah ok...just checking